diff -Naur -x '*.o' -x '*.cmd' pulseaudio-modules-bt-master/src/modules/bluetooth/module-bluez5-device.c patched/src/modules/bluetooth/module-bluez5-device.c
--- pulseaudio-modules-bt-master/src/modules/bluetooth/module-bluez5-device.c	2019-05-19 22:31:31.000000000 -0700
+++ patched/src/modules/bluetooth/module-bluez5-device.c	2019-06-29 21:59:46.267359417 -0700
@@ -1337,9 +1337,9 @@
     pa_log_debug("IO Thread starting up");
 
     if (u->core->realtime_scheduling)
-#ifdef PulseAudio_VERSION_11_N
+#ifdef PulseAudio_VERSION_11_N || PulseAudio_VERSION_12_N
         pa_make_realtime(u->core->realtime_priority);
-#elif  PulseAudio_VERSION_12_N
+#elif  PulseAudio_VERSION_13_N
         pa_thread_make_realtime(u->core->realtime_priority);
 #endif
     pa_thread_mq_install(&u->thread_mq);
